spec/system/signup_spec.rb

```diff
@@ -50,12 +55,43 @@
     expect(page).to have_text("Name can only contain letters and spaces")
   end
 
-  it "does sign-up when valid credentials" do
-    fill_in "Name", with: "user"
-    fill_in "Email", with: "user1@example.com"
-    fill_in "Password", with: "secret"
-    click_button "Sign up"
+  context "does sign-up when valid credentials" do
+    before do
+      allow(Flipper).to receive(:enabled?).with(:block_registration).and_return(false)
+      allow(Flipper).to receive(:enabled?).with(:recaptcha).and_return(false)
+      allow(Flipper).to receive(:enabled?).with(:gitlab_integration).and_return(false)
+      allow(Flipper).to receive(:enabled?).with(:forum).and_return(false)
+    end
+
+    it "does sign-up when valid credentials" do
+      fill_in "Name", with: "user"
+      fill_in "Email", with: "user1@example.com"
+      fill_in "Password", with: "secret"
+      click_button "Sign up"
+
+      expect(page).to have_text("Welcome! You have signed up successfully.")
+    end
+  end
+
+  context "when registration is disabled" do
+    before do
+      allow(Flipper).to receive(:enabled?).with(:block_registration).and_return(true)
+      allow(Flipper).to receive(:enabled?).with(:recaptcha).and_return(false)
+      allow(Flipper).to receive(:enabled?).with(:gitlab_integration).and_return(false)
+      allow(Flipper).to receive(:enabled?).with(:sso_integration).and_return(false)
+      allow(Flipper).to receive(:enabled?).with(:forum).and_return(false)
```

Comment:
We can disable this by default in spec_helper so that we don't need to duplicate these every specs.